mmengine/hooks/param_scheduler_hook.py

```diff
@@ -59,15 +61,45 @@ def after_train_epoch(self, runner) -> None:
             runner (Runner): The runner of the training process.
         """
 
+        if runner.param_schedulers is None:
+            return
+
         def step(param_schedulers):
             assert isinstance(param_schedulers, list)
             for scheduler in param_schedulers:
                 if scheduler.by_epoch:
                     scheduler.step()
 
+        if isinstance(runner.param_schedulers, list):
+            step(runner.param_schedulers)
+        elif isinstance(runner.param_schedulers, dict):
+            for param_schedulers in runner.param_schedulers.values():
+                step(param_schedulers)
+        else:
+            raise TypeError(
+                'runner.param_schedulers should be list of ParamScheduler or '
+                'a dict containing list of ParamScheduler, '
+                f'but got {runner.param_schedulers}')
+
+    def after_val_epoch(self,
+                        runner,
+                        metrics: Optional[Dict[str, float]] = None) -> None:
+
         if runner.param_schedulers is None:
             return
 
+        if not self._should_after_val_epoch(runner):
+            return
+
+        def step(param_schedulers):
+            assert isinstance(param_schedulers, list)
+            for scheduler in param_schedulers:
+                if isinstance(scheduler, ReduceOnPlateauParamScheduler):
+                    if scheduler.by_epoch and (scheduler.monitor_stage
+                                               == 'val'):
+                        scheduler.metric = metrics[scheduler.monitor_key]
+                        scheduler.step()
```

Comment:
Could we directly pass the metric to the `step` method, and parse it in the method?

Reply:
Ok, I will design the part again.